extensions/wikidata/src/org/openrefine/wikidata/editing/ReconEntityRewriter.java

```diff
@@ -69,12 +74,30 @@
 	 * @param subject
 	 *      the subject id of the entity to rewrite
 	 */
-	public ReconEntityRewriter(NewItemLibrary library, ItemIdValue subject) {
+	public ReconEntityRewriter(NewItemLibrary library, EntityIdValue subject) {
 		super(new DataObjectFactoryImpl());
 		this.library = library;
 		this.subject = subject;
 	}
-
+	
+	public EntityIdValue copy(EntityIdValue value) {
+		if (value instanceof ReconEntityIdValue) {
+			ReconEntityIdValue recon = (ReconEntityIdValue) value;
+			if (recon.isNew()) {
+				String newId = library.getQid(recon.getReconInternalId());
+				if (newId == null) {
+					if (subject.equals(recon)) {
+						return (EntityIdValue) subject;
+					} else {
+						throw new MissingEntityIdFound(recon);
+					}
+				}
+				return Datamodel.makeItemIdValue(newId, recon.getRecon().identifierSpace);
+			}
+		}
+		return super.copy((ItemIdValue) value);
+	}
```

Comment:
According to the error message, the failing cast seems to be happening here, and it makes sense because this line is indeed problematic: there is no reason why the `EntityIdValue` we are passed is an `ItemIdValue` in general.

I think this entire method can actually be removed now that you have added other methods for the other classes that we intend to support in this context.